app/index.js

```diff
@@ -1400,53 +1402,55 @@ module.exports = JhipsterGenerator.extend({
-            this.removefile(TEST_JS_DIR + 'spec/app/account/password/passwordControllerSpec.js');
-            this.removefile(TEST_JS_DIR + 'spec/app/account/password/passwordDirectiveSpec.js');
-            this.removefile(TEST_JS_DIR + 'spec/app/account/sessions/sessionsControllerSpec.js');
-            this.removefile(TEST_JS_DIR + 'spec/app/account/settings/settingsControllerSpec.js');
-            this.removefile(TEST_JS_DIR + 'spec/components/auth/authServicesSpec.js');
-
-            this.removefile('docker-compose.yml');
-            this.removefile('docker-compose-prod.yml');
-            this.removefile('Cassandra-Dev.Dockerfile');
-            this.removefile('Cassandra-Prod.Dockerfile');
-            this.removefolder('docker/');
-
-            this.removefile('gatling.gradle');
-            this.removefile('liquibase.gradle');
-            this.removefile('mapstruct.gradle');
-            this.removefile('profile_dev.gradle');
-            this.removefile('profile_fast.gradle');
-            this.removefile('profile_prod.gradle');
-            this.removefile('sonar.gradle');
-            this.removefile('yeoman.gradle');
+            if (this.installedVersionIsLessThan('2.27.1', this.jhipsterVersion)) {
+                this.removefile(javaDir + 'config/MailConfiguration.java');
+                this.removefile(javaDir + 'config/metrics/JavaMailHealthIndicator.java');
+                if (this.databaseType == 'sql' || this.databaseType == 'mongodb') {
+                    this.removefolder(javaDir + 'config/metrics');
+                }
+
+                this.removefile(javaDir + 'security/_CustomUserDetails.java');
+                this.removefile(javaDir + 'domain/util/CustomLocalDateSerializer.java');
+                this.removefile(javaDir + 'domain/util/CustomDateTimeSerializer.java');
+                this.removefile(javaDir + 'domain/util/CustomDateTimeDeserializer.java');
+                this.removefile(javaDir + 'domain/util/CustomLocalDateDeserializer.java');
+                this.removefile(javaDir + 'domain/util/DateToZonedDateTimeConverter.java');
+                this.removefile(javaDir + 'domain/util/ZonedDateTimeToDateConverter.java');
+                this.removefile(javaDir + 'domain/util/DateToLocalDateConverter.java');
+                this.removefile(javaDir + 'domain/util/LocalDateToDateConverter.java');
+                this.removefile(javaDir + 'domain/util/ISO8601LocalDateDeserializer.java');
+                this.removefolder(javaDir + 'web/propertyeditors');
+
+                this.removefile(RESOURCE_DIR + 'logback.xml');
+
+                this.removefile(WEBAPP_DIR + 'scripts/app/account/logout/logout.js');
+                this.removefile(WEBAPP_DIR + 'scripts/app/account/logout/logout.controller.js');
+                this.removefolder(WEBAPP_DIR + 'scripts/app/account/logout');
+
+                this.removefile(testDir + 'config/MongoConfiguration.java');
```

Comment:
some of these are added for very old versions so might be time to remove these now, @jdubois may be we should come up with a policy on how many backward versions to do this for and then remove the file remove instruction from code else this will get piled on